lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1082,6 +1095,40 @@ def get_scsi_type(self):
                     self.module.fail_json(msg="hardware.scsi attribute should be 'paravirtual' or 'lsilogic'")
         return disk_controller_type
 
+    def find_folder(self, searchpath):
+        """ Walk inventory objects one position of the searchpath at a time """
+
+        # split the searchpath so we can iterate through it
+        paths = [x.replace('/', '') for x in searchpath.split('/')]
+        paths = [x.strip() for x in paths if x.strip()]
```

Comment:
* ```x.replace('/', '')``` shouldn't be necessary.  searchpath.split('/') should have removed all instances of "/" from the output.
* Is it valid to strip the path elements?  Could you have a valid path like this: ```/folder1/  datacenter1  /vm```?